src/test/ui/editions/edition-deny-async-fns-2015.rs

```diff
@@ -0,0 +1,19 @@
+// edition:2015
+
+#![feature(futures_api)]
+
+async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition
+                  //~^ ERROR async fn is unstable
+
+struct Foo {}
+
+impl Foo {
+    async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition
+}
+
+trait Bar {
+    async fn foo() {} //~ ERROR `async fn` is not permitted in the 2015 edition
+                      //~^ ERROR trait fns cannot be declared `async`
+}
```

Comment:
Can you add the following?

```rust
macro_rules! accept_item { ($x:item) => {} }

accept_item! {
   async fn foo() {}
}
```
This checks that async fns are not *syntactically valid* on 2015 (which they shouldn't be).

Reply:
I've added this to the test, and it doesn't error... Is that the desired behaviour?